templates/unnest_parquet_view.sql.py

```diff
@@ -0,0 +1,154 @@
+                and field["fields"][0]["mode"] == "REPEATED"
+                and len(field["fields"][0]["fields"]) == 1
+                and field["fields"][0]["fields"][0]["name"] == "element"
+            ):
+                unnest = ".".join(prefix + (field["name"], "list"))
+                nested = replace(
+                    field["fields"][0]["fields"][0], unnest_layer + 1, unnest_as
+                )
+                # handle simplest case without unnest_as
+                if nested == f"{unnest_as}.element":
+                    return f"ARRAY(SELECT * FROM UNNEST({unnest})) AS {field['name']}"
+            # handle maps
+            elif (
+                field["fields"][0]["name"] == "key_value"
+                and field["fields"][0]["mode"] == "REPEATED"
+            ):
+                unnest = ".".join(prefix + (field["name"], "key_value"))
+                nested = replace(
+                    {
+                        "fields": field["fields"][0]["fields"],
+                        "name": unnest_as,
+                        "type": "RECORD",
+                    },
+                    unnest_layer + 1,
+                )
+                # handle simplest case without array unnest_layer
+                if nested == unnest_as:
+                    return f"{unnest} AS {field['name']}"
+            # shorten special case for nested struct
+            special_prefix = f"(SELECT AS STRUCT {unnest_as}.* "
+            if nested.startswith(special_prefix):
+                index = len(special_prefix)
+                suffix = nested[index:].rsplit(")", 1)[0]
+                select = f"SELECT AS STRUCT * {suffix}"
+            # shorten nested struct
+            elif nested.startswith("(SELECT AS STRUCT"):
+                select = nested[1:].rsplit(")", 1)[0]
+            # handle nested array
+            elif nested.startswith("ARRAY"):
+                select = f"SELECT AS STRUCT {nested}"
+            else:
+                select = f"SELECT {nested}"
+            from_ = f"FROM UNNEST({unnest}) AS {unnest_as}"
+            return f"ARRAY({select} {from_}) AS {field['name']}"
+        # handle other structs with nested replacements
+        replacements = []
+        for subfield in field["fields"]:
+            replacement = replace(subfield, unnest_layer, *prefix, field["name"])
+            if " AS " in replacement:
+                replacements += [replacement]
+        if replacements:
+            select = f"SELECT AS STRUCT {'.'.join(prefix + (field['name'], '*'))}"
+            replacements = ", ".join(replacements)
+            return f"({select} REPLACE ({replacements})) AS {field['name']}"
+    # no unnesting needed
+    return ".".join(prefix + (field["name"],))
+
+
+def generate(table, view):
+    """Generate a CREATE VIEW statement that unnests lists and maps."""
+    result = subprocess.check_output(
+        ["bq", "show", "--schema", table.replace(".", ":", 1)]
```

Comment:
Actually, I realized one thing after the fact -- is there a reason to prefer this over using the client libraries? seems the pros for the library is that the interface should remain stable for the version that we're using, while the cli output could change out from under us

Reply:
i'm fine using the client libs.

it's faster for me during development to call out to the cli, and it's done in other places in this repo, which is why i did this.

using the client libs requires they be installed on the python path, which isn't necessarily true if someone only installed the gcloud cli tools and expected it to work. then again this can be run in the docker just fine.

the client libs don't expose some things via a public method, as seen [in mozilla/parquet2bigquery#29](https://github.com/mozilla/parquet2bigquery/pull/29/files#diff-b6903abb1102a1109ac4bffcfc06ce64R207), but that doens't apply here because this is a table schema not a query schema

Reply:
okay, using python sdk now